blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/scm/Scm.java

```diff
@@ -73,4 +79,34 @@
     @PUT
     @WebMethod(name = VALIDATE)
     public abstract @CheckForNull HttpResponse validateAndCreate(@JsonBody JSONObject request);
+
+    /**
+     * Save a file to this repository. Creates a new one if it doesn't exist.
+     *
+     * @return {@link ScmFile}
+     */
+    @PUT
+    @WebMethod(name="")
+    @TreeResponse
+    public ScmFile saveFile(StaplerRequest staplerRequest) throws IOException {
+        JSONObject body = JSONObject.fromObject(IOUtils.toString(staplerRequest.getReader()));
+        String cls = (String) body.get("$class");
+        if(cls == null){
+            throw new ServiceException.BadRequestExpception(new ErrorMessage(400, "Failed to save file")
+                    .add(new ErrorMessage.Error("$class", ErrorMessage.Error.ErrorCodes.MISSING.toString(),
+                            "$class is required")));
+        }
+        ScmSaveFileRequest request = staplerRequest.bindJSON(ScmSaveFileRequest.class, body);
```

Comment:
Seems like this should be dirt simple to provide a couple annotations to just automatically do the bindings in and out, without needing `@Exported...` on things, maybe it's just a preference, but I prefer service methods to reference the objects they use with clear signatures. It pains me when I have to explain to people how to write REST services currently, but that's a topic for another day, nothing to do with this PR.